jobserver/views/workspaces.py

```diff
@@ -318,6 +294,44 @@ def get(self, request, *args, **kwargs):
             context=context,
         )
 
+    def get_output_permissions(self, user, workspace):
+        is_privileged_user = has_permission(
+            user, "release_file_view", project=workspace.project
+        )
+
+        # a user can see backend files if they have access to at least one
+        # backend and the permissions required to see outputs
+        has_backends = (
+            user.is_authenticated and user.backends.exclude(level_4_url="").exists()
+        )
+        can_view_files = is_privileged_user and has_backends
+
+        # are there any releases to show for the workspace?
+        can_view_releases = workspace.releases.exists()
+
+        # unprivileged users can only see published snapshots, but privileged
+        # users can see snapshots if there are any releases since they can also
+        # prepare and publish them from the same views.
+        has_published_snapshots = workspace.snapshots.exclude(
+            published_at=None
+        ).exists()
+        has_any_snapshots = workspace.snapshots.exists()
+        can_view_outputs = has_published_snapshots or (
+            is_privileged_user and (can_view_releases or has_any_snapshots)
+        )
+
+        return {
+            "level_4": {
+                "disabled": not can_view_files,
```

Comment:
This might be one for a future PR, but it might be helpful to show the user whether the link is disabled because they don't have permission or because there aren't any files.